src/runners/rfq_maker_balance_cache_runner.ts

```diff
@@ -0,0 +1,222 @@
+        const web3Wrapper = new Web3Wrapper(provider);
+
+        const connection = await getDBConnectionAsync();
+        const balanceCheckerContractInterface = getBalanceCheckerContractInterface(RANDOM_ADDRESS, provider);
+
+        await runRfqBalanceCacheAsync(web3Wrapper, connection, balanceCheckerContractInterface);
+    })().catch(error => logger.error(error.stack));
+}
+
+async function runRfqBalanceCacheAsync(
+    web3Wrapper: Web3Wrapper,
+    connection: Connection,
+    balanceCheckerContractInterface: BalanceCheckerContract,
+): Promise<void> {
+    const workerId = _.uniqueId('rfqw_');
+    let lastBlockSeen = -1;
+    while (true) {
+        const newBlock = await web3Wrapper.getBlockNumberAsync();
+        if (lastBlockSeen < newBlock) {
+            lastBlockSeen = newBlock;
+            LATEST_BLOCK_PROCESSED_GAUGE.labels(workerId).set(lastBlockSeen);
+            logUtils.log(
+                {
+                    block: lastBlockSeen,
+                    workerId,
+                },
+                'Found new block',
+            );
+
+            await cacheRfqBalancesAsync(connection, balanceCheckerContractInterface, true);
+
+            await delay(DELAY_WHEN_NEW_BLOCK_FOUND);
+        } else {
+            await delay(DELAY_WHEN_NEW_BLOCK_NOT_FOUND);
+        }
+    }
+}
+
+/**
+ * This function retrieves and caches ERC20 balances of RFQ market makers
+ */
+export async function cacheRfqBalancesAsync(
+    connection: Connection,
+    balanceCheckerContractInterface: BalanceCheckerContract,
+    codeOverride: boolean,
+): Promise<void> {
+    const makerTokens = await getMakerTokensAsync(connection);
+    const balancesCallInput = splitValues(makerTokens);
+
+    const updateTime = new Date();
+    const erc20Balances = await getErc20BalancesAsync(balanceCheckerContractInterface, balancesCallInput, codeOverride);
+
+    await updateErc20BalancesAsync(balancesCallInput, erc20Balances, connection, updateTime);
+}
+
+// NOTE: this only returns a partial entity class, just token address and maker address
+// Cache the query results to reduce reads from the DB
+let MAKER_TOKEN_CACHE: ResultCache<MakerBalanceChainCacheEntity[]>;
+async function getMakerTokensAsync(connection: Connection): Promise<MakerBalanceChainCacheEntity[]> {
+    const start = new Date().getTime();
+
+    if (!MAKER_TOKEN_CACHE) {
+        MAKER_TOKEN_CACHE = createResultCache<any[]>(
+            () =>
+                connection
+                    .getRepository(MakerBalanceChainCacheEntity)
+                    .createQueryBuilder('maker_balance_chain_cache')
+                    .select(['maker_balance_chain_cache.tokenAddress', 'maker_balance_chain_cache.makerAddress'])
+                    .getMany(),
+            CACHE_MAKER_TOKENS_FOR_MS,
+        );
+    }
+    const results = (await MAKER_TOKEN_CACHE.getResultAsync()).result;
+
+    MAKER_BALANCE_CACHE_RESULT_COUNT.set(results.length);
```

Comment:
Just to keep things consistent, would you mind adding the same label here too? thanks

Reply:
👍